fms_extras/modules/speculator.py

```diff
@@ -0,0 +1,91 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from fms.modules.layernorm import LayerNormParameterized
+
+class Speculator(nn.Module):
```

Comment:
I wonder if we might end up with multiple speculators, and if there's something more specific we might want to name this

Reply:
might be helpful to have an explanation of our speculation strategy in the comments too

Reply:
Good point, I'll add an explanation. We could call this MLP_Speculator

Reply:
FYI snake_case class names aren't convention in python: https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html

we might want to rename the dataset classes at some point too